crates/polars-ops/src/series/ops/business.rs

```diff
@@ -7,14 +8,28 @@ use polars_core::prelude::*;
 /// - `start`: Series holding start dates.
 /// - `end`: Series holding end dates.
 /// - `week_mask`: A boolean array of length 7, where `true` indicates that the day is a business day.
+/// - `holidays`: timestamps that are holidays. Must be provided as i32, i.e. the number of
+///   days since the UNIX epoch.
 pub fn business_day_count(
     start: &Series,
     end: &Series,
     week_mask: [bool; 7],
+    holidays: &[i32],
 ) -> PolarsResult<Series> {
     if !week_mask.iter().any(|&x| x) {
         polars_bail!(ComputeError:"`week_mask` must have at least one business day");
     }
+
+    // De-dupe and sort holidays, and exclude non-business days.
+    let mut holidays: Vec<i32> = holidays
+        .iter()
+        .filter(|&x| *unsafe { week_mask.get_unchecked(weekday(*x)) })
+        .cloned()
+        .collect::<HashSet<_>>()
+        .into_iter()
+        .collect();
+    holidays.sort_unstable();
```

Comment:
Sorting and unique feel very similar operations. I feel that can be in one.

Reply:
Thanks for your review!

What numpy do is:
1. sort
2. de-dupe + filter out weekends in the same step

I wouldn't know how to combine these further - but note that this operation only needs doing once per `business_day_count` call (it's outside the loop over rows) and that, unless people are extraordinarily lucky, they're probably only passing in a a limited number of holiday days